src/components/TopBar/ReactionMenu.vue

```diff
@@ -0,0 +1,176 @@
+				fill-color="#ffffff" />
+		</template>
+
+		<NcActionButtonGroup class="reaction__group"
+			:style="{'--reactions-in-single-row': reactionsInSingleRow}">
+			<NcActionButton v-for="(reaction, index) in reactions"
+				:key="index"
+				:aria-label="t('spreed', 'React with {reaction}', { reaction })"
+				class="reaction__button"
+				@click="throttledSendReaction(reaction)">
+				<template #icon>
+					{{ reaction }}
+				</template>
+			</NcActionButton>
+		</NcActionButtonGroup>
+
+		<!-- TODO for development purposes, remove before release -->
+		<NcActionButton @click="spamReaction">
+			{{ spammer ? 'Stop spamming' : 'Spam reactions' }}
+		</NcActionButton>
+	</NcActions>
+</template>
+
+<script>
+import EmoticonOutline from 'vue-material-design-icons/EmoticonOutline.vue'
+
+import { emit } from '@nextcloud/event-bus'
+
+import NcActionButton from '@nextcloud/vue/dist/Components/NcActionButton.js'
+import NcActionButtonGroup from '@nextcloud/vue/dist/Components/NcActionButtonGroup.js'
+import NcActions from '@nextcloud/vue/dist/Components/NcActions.js'
+
+export default {
+	name: 'ReactionMenu',
+
+	components: {
+		NcActions,
+		NcActionButton,
+		NcActionButtonGroup,
+		EmoticonOutline,
+	},
+
+	props: {
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		 * Signalling participant model
+		 */
+		localCallParticipantModel: {
+			type: Object,
+			required: true,
+		},
+
+		/**
+		 * Supported reactions
+		 */
+		reactions: {
+			type: Array,
+			validator: (prop) => prop.every(e => typeof e === 'string'),
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			throttleTimer: null,
+			// TODO for development purposes, remove before release
+			spammer: null,
+		}
+	},
+
+	computed: {
+		container() {
+			return this.$store.getters.getMainContainerSelector()
+		},
+
+		reactionsInSingleRow() {
+			return Math.ceil(this.reactions.length / 2)
+		},
+	},
+
+	methods: {
+		throttledSendReaction(reaction) {
+			if (this.throttleTimer) {
+				return
+			}
+
+			this.sendReaction(reaction)
+			this.throttleTimer = setTimeout(() => {
+				this.throttleTimer = null
+			}, 2000)
+		},
+
+		sendReaction(reaction) {
+			// send reaction to other participants
+			this.localCallParticipantModel.sendReaction(reaction)
+
+			// show reaction to yourself
+			emit('send-reaction', {
+				model: this.localCallParticipantModel,
+				reaction,
+			})
+		},
+
+		// TODO for development purposes, remove before release
+		spamReaction() {
+			if (this.spammer) {
+				clearInterval(this.spammer)
+				this.spammer = null
+				return
+			}
+
+			this.spammer = setInterval(() => {
+				const reactionRand = this.reactions[Math.floor(Math.random() * 10)]
+				this.localCallParticipantModel.sendReaction(reactionRand)
+
+				// show reaction to yourself
+				emit('send-reaction', {
+					model: this.localCallParticipantModel,
+					reaction: reactionRand,
+				})
+			}, 1000)
+		},
```

Comment:
**thought:** we may keep this method and only remove the button from UI

Reply:
Then could it be used by someone through the devtools? Or it will not be accessible in production?

Reply:
I'd like to remove it at all, as it's not hard to reproduce again. Or at least move this implementation in unit test, there it could be used for good

Reply:
https://github.com/nextcloud/spreed/blob/788ec820eb612b706d5b70b8c70b2494730dcfb5/src/components/NewMessageForm/NewMessageForm.vue#L533

Maybe make sure to wrap it with OC.debug :P

Reply:
> Maybe make sure to wrap it with OC.debug :P

In context of "how easy is it to run a testing method", it is still very easy to set `OC.debug` to `true`